frontend/src/component/changeRequest/ChangeRequestOverview/ReviewButton/ReviewButton.tsx

```diff
@@ -0,0 +1,135 @@
+    const anchorRef = React.useRef<HTMLButtonElement>(null);
+
+    const onMenuItemClick = async (index: number) => {
+        try {
+            const option = options[index];
+            if (option.includes('Approve')) {
+                await changeState(projectId, Number(id), { state: 'Approved' });
+                refetchChangeRequest();
+                setToastData({
+                    type: 'success',
+                    title: 'Success',
+                    text: 'Changes approved',
+                });
+            }
+
+            if (option.includes('Reject')) {
+                await changeState(projectId, Number(id), {
+                    state: 'Cancelled',
+                });
+                refetchChangeRequest();
+                setToastData({
+                    type: 'success',
+                    title: 'Success',
+                    text: 'Changes rejected',
+                });
+            }
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        }
+    };
+
+    const onToggle = () => {
+        setOpen(prevOpen => !prevOpen);
+    };
+
+    const onClose = (event: Event) => {
+        if (
+            anchorRef.current &&
+            anchorRef.current.contains(event.target as HTMLElement)
+        ) {
+            return;
+        }
+
+        setOpen(false);
+    };
+
+    return (
+        <React.Fragment>
+            <Button
+                variant="contained"
+                aria-controls={open ? 'review-options-menu' : undefined}
+                aria-expanded={open ? 'true' : undefined}
+                aria-label="review changes"
+                aria-haspopup="menu"
+                onClick={onToggle}
+                ref={anchorRef}
+                endIcon={<ArrowDropDownIcon />}
+            >
+                Review changes
+            </Button>
+            <Popper
+                sx={{
+                    zIndex: 1,
+                }}
+                open={open}
+                anchorEl={anchorRef.current}
+                role={undefined}
+                transition
+                disablePortal
+            >
+                {({ TransitionProps, placement }) => (
+                    <Grow
+                        {...TransitionProps}
+                        style={{
+                            transformOrigin:
+                                placement === 'bottom'
+                                    ? 'center top'
+                                    : 'center bottom',
+                        }}
+                    >
+                        <Paper>
+                            <ClickAwayListener onClickAway={onClose}>
+                                <MenuList
+                                    id="review-options-menu"
+                                    autoFocusItem
+                                >
+                                    {options.map((option, index) => (
```

Comment:
I'd simplify it by passing two explicit MenuItem components each with its own onMenuItemClick. Currently to track the logic I had to iterate over two strings and then visually filter out a correct path inside onMenuItemClick.

Reply:
Thanks! That is much better indeed. I got locked into material-uis example here.